iac-extensions/terraform/src/test/resources/checks/Azure/SubscriptionOwnerCapabilitiesCheck/noncompliant.tf

```diff
@@ -0,0 +1,113 @@
+resource "azurerm_role_definition" "subscription-owner-role" {
+  name        = "subscription-owner-role"
+  scope       = data.azurerm_subscription.primary.id
+
+  permissions {
+    actions     = ["*"]
+    not_actions = []
+  }
+
+  assignable_scopes = [
+    data.azurerm_subscription.primary.id
+  ]
+}
+
+resource "azurerm_role_definition" "subscription-owner-role-multiple-scopes" { # Noncompliant
+  name        = "subscription-owner-role"
+  scope       = data.azurerm_subscription.primary.id
+
+  permissions {
+    actions     = ["*"]
+    not_actions = []
+  }
+
+  assignable_scopes = [
+    data.azurerm_subscription.foo.id,
+    data.azurerm_subscription.primary.id
+  ]
+}
+
+
+# Alternative ways to define a custom role with a subscription assignable scope
+resource "azurerm_role_definition" "subscription-owner-role-alternative-scope-1" { # Noncompliant
+  name        = "subscription-owner-role-alternative-scope-1"
+  scope       = data.azurerm_subscription.primary.id
+  permissions {
+    actions     = ["*"]
+    not_actions = []
+  }
+
+  assignable_scopes = [
+    "${data.azurerm_subscription.primary.id}"
+  ]
+}
+
+
+resource "azurerm_role_definition" "subscription-owner-role-alternative-scope-2" { # Noncompliant
+  name        = "subscription-owner-role-alternative-scope-2"
+  scope       = data.azurerm_subscription.primary.id
+  permissions {
+    actions     = ["*"]
+    not_actions = []
+  }
+
+  assignable_scopes = [
+    "/subscriptions/00000000-0000-0000-0000-000000000000"
+  ]
+}
+
+resource "azurerm_role_definition" "subscription-owner-role-alternative-scope-3" { # FN - We do not resolve variables yet
+  name        = "subscription-owner-role-alternative-scope-3"
+  scope       = data.azurerm_subscription.primary.id
+
+  permissions {
+    actions     = ["*"]
+    not_actions = []
+  }
+
+  assignable_scopes = [
+    "/subscriptions/${var.subscription_id}/"
+  ]
+}
+
+# Custom role defined with a Management Group assignable scope
+resource "azurerm_role_definition" "management-group-owner-role" { # Noncompliant
+  name        = "management group-owner-role"
+  scope       = data.azurerm_management_group.example_parent.id
+
+  permissions {
+    actions     = ["*"]
+    not_actions = []
+  }
+
+  assignable_scopes = [
+    data.azurerm_management_group.example_parent.id
+  ]
+}
+
+# Custom role defined with a Management Group assignable scope
+resource "azurerm_role_definition" "management-group-owner-role" { # Noncompliant
+  permissions {
+    actions     = ["*"]
+  }
+
+  assignable_scopes = [
+    data.azurerm_management_group.root_example.id
+  ]
+}
+
+# Alternative way to define a custom role with a Management Group assignable scope
+resource "azurerm_role_definition" "management-group-owner-role-alternative-scope" { # Noncompliant
+  name        = "management-group-owner-role"
+  scope       = data.azurerm_management_group.example_parent.id
```

Comment:
Where's the data definition of this data reference, or have I understood data incorrectly?

Reply:
You've properly understood this correctly. However, for this check, we do not need to define the data block because we don't consider if the block truly exists. When you want to see real executable code, you can have a look at the "[expected issues](https://github.com/SonarSource/security-expected-issues/pull/626/files)" for each rule. The AppSec bubble tries to test the code in a real environment. Nevertheless, for our parser as well as our checks we do not need to write runnable code.